libpod/image/image.go

```diff
@@ -1475,9 +1476,10 @@ func (i *Image) Save(ctx context.Context, source, format, output string, moreTag
 			return errors.Wrapf(err, "error getting OCI archive ImageReference for (%q, %q)", output, destImageName)
 		}
 	case "oci-dir":
-		destRef, err = directory.NewReference(output)
+		destImageName := imageNameForSaveDestination(i, source)
+		destRef, err = layout.NewReference(output, destImageName)
 		if err != nil {
-			return errors.Wrapf(err, "error getting directory ImageReference for %q", output)
+			return errors.Wrapf(err, "error saving the oci directory ImageReference for (%q, %q)", output, destImageName)
```

Comment:
This is not quite right — actual saving happens in `PushImageToReference` below.
```suggestion
			return errors.Wrapf(err, "error getting OCI directory ImageReference for (%q, %q)", output, destImageName)
```